test/docker-e2e/dockerchain/testchain.go

```diff
@@ -0,0 +1,119 @@
+package dockerchain
+
+import (
+	"context"
+	"github.com/celestiaorg/celestia-app/v4/app"
+	"github.com/celestiaorg/celestia-app/v4/app/encoding"
+	"github.com/celestiaorg/celestia-app/v4/pkg/user"
+	"github.com/celestiaorg/celestia-app/v4/test/util/genesis"
+	tastoradockertypes "github.com/celestiaorg/tastora/framework/docker"
+	"github.com/celestiaorg/tastora/framework/testutil/config"
+	"github.com/celestiaorg/tastora/framework/testutil/maps"
+	cometcfg "github.com/cometbft/cometbft/config"
+	cmtjson "github.com/cometbft/cometbft/libs/json"
+	"github.com/cometbft/cometbft/privval"
+	servercfg "github.com/cosmos/cosmos-sdk/server/config"
+	"github.com/cosmos/cosmos-sdk/types/module/testutil"
+	"github.com/stretchr/testify/require"
+	"testing"
+	"time"
+)
+
+// NewCelestiaChainBuilder constructs a new ChainBuilder configured for a Celestia instance with predefined parameters.
+func NewCelestiaChainBuilder(t *testing.T, cfg *Config) *tastoradockertypes.ChainBuilder {
```

Comment:
the idea with this function is that it would provide the "default" way that a celestia chain is built, from the node repo, the E2E tests would call this function.